ledger/test-common/src/main/scala/com/digitalasset/platform/testing/TimeBoundObserver.scala

```diff
@@ -10,19 +10,24 @@ import io.grpc.stub.StreamObserver
 import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.{ExecutionContext, Future, Promise}
 
-final class TimeBoundObserver[T](duration: FiniteDuration)(
+final class TimeBoundObserver[T](duration: FiniteDuration, maximumResults: Option[Int] = None)(
```

Comment:
Not necessary, but I would recommend adding a secondary constructor that takes an `Int` and wraps it into an option (pure UX).

Reply:
If we use this more than once, I'd be happy to add it.